File: backend/app/routers/auto_reply.py
"""Auto-reply router."""

from fastapi import APIRouter, HTTPException
from pydantic import BaseModel

from app.schemas.automation import AutoReplyConfig, AutoReplyResponse
from app.services.auto_reply_service import should_auto_reply, generate_reply

router = APIRouter()

# In-memory config store (use DB in production)
_configs: dict[str, AutoReplyConfig] = {}


class AutoReplyTestRequest(BaseModel):
    """Request model for testing auto-reply."""
    comment_text: str
    platform: str
    tone: str = "professional"


@router.post("/config", response_model=AutoReplyResponse)
async def set_auto_reply_config(config: AutoReplyConfig):
    """Configure auto-reply settings."""
    key = f"{config.workspace_id}:{config.platform}"
    _configs[key] = config
    return AutoReplyResponse(
        id=key,
        workspace_id=config.workspace_id,
        platform=config.platform,
        is_enabled=config.is_enabled,
        reply_tone=config.reply_tone,
        max_replies_per_day=config.max_replies_per_day,
    )


@router.post("/test")
async def test_auto_reply(request: AutoReplyTestRequest):
    """Test auto-reply generation."""
    should_reply = await should_auto_reply(request.platform, request.comment_text)
    if not should_reply:
        return {"should_reply": False, "reply": None}

    reply = await generate_reply(request.comment_text, request.platform, request.tone)
    return {"should_reply": True, "reply": reply}

File: frontend/src/services/auto-reply.ts
import api from "@/lib/api";

export interface AutoReplyConfig {
  workspace_id: string;
  platform: string;
  is_enabled: boolean;
  reply_tone: string;
  max_replies_per_day: number;
  sentiment_threshold: number;
  excluded_keywords: string[];
}

export interface AutoReplyActivity {
  id: string;
  platform: string;
  comment_text: string;
  reply_text: string | null;
  sentiment_score: number;
  created_at: string;
}

export interface AutoReplyStats {
  total_replies_today: number;
  average_sentiment: number;
  is_enabled: boolean;
}

export async function testAutoReply(
  commentText: string,
  platform: string,
  tone: string = "professional"
): Promise<{ should_reply: boolean; reply: string | null }> {
  const res = await api.post(
    `/auto-reply/test?comment_text=${encodeURIComponent(commentText)}&platform=${platform}&tone=${tone}`
  );
  return res.data;
}

export async function saveAutoReplyConfig(config: AutoReplyConfig) {
  const res = await api.post("/auto-reply/config", config);
  return res.data;
}

export async function getAutoReplyStats(workspaceId: string): Promise<AutoReplyStats> {
  const res = await api.get<AutoReplyStats>(`/auto-reply/stats?workspace_id=${workspaceId}`);
  return res.data;
}

export async function getAutoReplyActivity(workspaceId: string): Promise<AutoReplyActivity[]> {
  const res = await api.get<AutoReplyActivity[]>(`/auto-reply/activity?workspace_id=${workspaceId}`);
  return res.data;
}
